### ISSUE TITLE
[BUG] firmwarecheck uses incorrect URL (http://https://repo.shelly.cloud/files/firmware)

### ISSUE THREAD

**Issue by wivaku** [StyraHem/ShellyForHASS#460]
## Environment

- ShellyForHASS release: 0.2.0
- HA core: 0.116.2
- operating environment: Virtualbox virtual appliance 

## Describe the bug

Log shows incorrect URL for firmware check:

`Error http GET: http://https://repo.shelly.cloud/files/firmware `
(note the `http://https://...`)

## Steps to Reproduce

No steps to perform

## Expected behavior

No error (by using the url `https://repo.shelly.cloud/files/firmware`)

**Comment by hakana** [comment]
This is just an error in the log function, it connecting to the right URL.

Do you got this error often?

**Comment by wivaku** [comment]
I have seen it multiple times in the log.
Currently I only have the log starting today, so I can not check how often it happens.
It triggers a long list of linked error messages.

**Comment by chpego** [comment]
#443

**Comment by david-kalbermatten** [comment]
Same here
```log
Logger: pyShelly
Source: /usr/local/lib/python3.8/site-packages/pyShelly/utils.py:38
First occurred: October 30, 2020, 2:37:27 PM (1 occurrences)
Last logged: October 30, 2020, 2:37:27 PM

Error http GET: http://https://repo.shelly.cloud/files/firmware, <urlopen error [Errno -3] Try again>, Traceback (most recent call last): File "/usr/local/lib/python3.8/urllib/request.py", line 1350, in do_open h.request(req.get_method(), req.selector, req.data, headers, File "/usr/local/lib/python3.8/http/client.py", line 1255, in request self._send_request(method, url, body, headers, encode_chunked) File "/usr/local/lib/python3.8/http/client.py", line 1301, in _send_request self.endheaders(body, encode_chunked=encode_chunked) File "/usr/local/lib/python3.8/http/client.py", line 1250, in endheaders self._send_output(message_body, encode_chunked=encode_chunked) File "/usr/local/lib/python3.8/http/client.py", line 1010, in _send_output self.send(msg) File "/usr/local/lib/python3.8/http/client.py", line 950, in send self.connect() File "/usr/local/lib/python3.8/http/client.py", line 1417, in connect super().connect() File "/usr/local/lib/python3.8/http/client.py", line 921, in connect self.sock = self._create_connection( File "/usr/local/lib/python3.8/socket.py", line 787, in create_connection for res in getaddrinfo(host, port, 0, SOCK_STREAM): File "/usr/local/lib/python3.8/socket.py", line 918, in getaddrinfo for res in _socket.getaddrinfo(host, port, family, type, proto, flags): socket.gaierror: [Errno -3] Try again During handling of the above exception, another exception occurred: Traceback (most recent call last): File "/usr/local/lib/python3.8/site-packages/pyShelly/firmware.py", line 29, in _http_get f = urllib.request.urlopen(url) File "/usr/local/lib/python3.8/urllib/request.py", line 222, in urlopen return opener.open(url, data, timeout) File "/usr/local/lib/python3.8/urllib/request.py", line 525, in open response = self._open(req, data) File "/usr/local/lib/python3.8/urllib/request.py", line 542, in _open result = self._call_chain(self.handle_open, protocol, protocol + File "/usr/local/lib/python3.8/urllib/request.py", line 502, in _call_chain result = func(*args) File "/usr/local/lib/python3.8/urllib/request.py", line 1393, in https_open return self.do_open(http.client.HTTPSConnection, req, File "/usr/local/lib/python3.8/urllib/request.py", line 1353, in do_open raise URLError(err) urllib.error.URLError: <urlopen error [Errno -3] Try again>
Traceback (most recent call last):
  File "/usr/local/lib/python3.8/urllib/request.py", line 1350, in do_open
    h.request(req.get_method(), req.selector, req.data, headers,
  File "/usr/local/lib/python3.8/http/client.py", line 1255, in request
    self._send_request(method, url, body, headers, encode_chunked)
  File "/usr/local/lib/python3.8/http/client.py", line 1301, in _send_request
    self.endheaders(body, encode_chunked=encode_chunked)
  File "/usr/local/lib/python3.8/http/client.py", line 1250, in endheaders
    self._send_output(message_body, encode_chunked=encode_chunked)
  File "/usr/local/lib/python3.8/http/client.py", line 1010, in _send_output
    self.send(msg)
  File "/usr/local/lib/python3.8/http/client.py", line 950, in send
    self.connect()
  File "/usr/local/lib/python3.8/http/client.py", line 1417, in connect
    super().connect()
  File "/usr/local/lib/python3.8/http/client.py", line 921, in connect
    self.sock = self._create_connection(
  File "/usr/local/lib/python3.8/socket.py", line 787, in create_connection
    for res in getaddrinfo(host, port, 0, SOCK_STREAM):
  File "/usr/local/lib/python3.8/socket.py", line 918, in getaddrinfo
    for res in _socket.getaddrinfo(host, port, family, type, proto, flags):
socket.gaierror: [Errno -3] Try again

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/usr/local/lib/python3.8/site-packages/pyShelly/firmware.py", line 29, in _http_get
    f = urllib.request.urlopen(url)
  File "/usr/local/lib/python3.8/urllib/request.py", line 222, in urlopen
    return opener.open(url, data, timeout)
  File "/usr/local/lib/python3.8/urllib/request.py", line 525, in open
    response = self._open(req, data)
  File "/usr/local/lib/python3.8/urllib/request.py", line 542, in _open
    result = self._call_chain(self.handle_open, protocol, protocol +
  File "/usr/local/lib/python3.8/urllib/request.py", line 502, in _call_chain
    result = func(*args)
  File "/usr/local/lib/python3.8/urllib/request.py", line 1393, in https_open
    return self.do_open(http.client.HTTPSConnection, req,
  File "/usr/local/lib/python3.8/urllib/request.py", line 1353, in do_open
    raise URLError(err)
urllib.error.URLError: <urlopen error [Errno -3] Try again>
```

**Comment by fatt0ny** [comment]
Getting similar errors filling logs!

2020-11-22 01:38:02 ERROR (FirmwareManage) [pyShelly] Error http GET: http://https://repo.shell>
Nov 22 01:38:02 Xserve 6b7568aaea15[585]: File "/usr/local/lib/python3.8/urllib/request.py", line 1350, in do_open
Nov 22 01:38:02 Xserve 6b7568aaea15[585]: h.request(req.get_method(), req.selector, req.data, headers,
Nov 22 01:38:02 Xserve 6b7568aaea15[585]: File "/usr/local/lib/python3.8/http/client.py", line 1255, in request
Nov 22 01:38:02 Xserve 6b7568aaea15[585]: self._send_request(method, url, body, headers, encode_chunked)
Nov 22 01:38:02 Xserve 6b7568aaea15[585]: File "/usr/local/lib/python3.8/http/client.py", line 1301, in _send_request
Nov 22 01:38:02 Xserve 6b7568aaea15[585]: self.endheaders(body, encode_chunked=encode_chunked)
Nov 22 01:38:02 Xserve 6b7568aaea15[585]: File "/usr/local/lib/python3.8/http/client.py", line 1250, in endheaders
Nov 22 01:38:02 Xserve 6b7568aaea15[585]: self._send_output(message_body, encode_chunked=encode_chunked)
Nov 22 01:38:02 Xserve 6b7568aaea15[585]: File "/usr/local/lib/python3.8/http/client.py", line 1010, in _send_output
Nov 22 01:38:02 Xserve 6b7568aaea15[585]: self.send(msg)
Nov 22 01:38:02 Xserve 6b7568aaea15[585]: File "/usr/local/lib/python3.8/http/client.py", line 950, in send
Nov 22 01:38:02 Xserve 6b7568aaea15[585]: self.connect()
Nov 22 01:38:02 Xserve 6b7568aaea15[585]: File "/usr/local/lib/python3.8/http/client.py", line 1417, in connect
Nov 22 01:38:02 Xserve 6b7568aaea15[585]: super().connect()
Nov 22 01:38:02 Xserve 6b7568aaea15[585]: File "/usr/local/lib/python3.8/http/client.py", line 921, in connect
Nov 22 01:38:02 Xserve 6b7568aaea15[585]: self.sock = self._create_connection(
Nov 22 01:38:02 Xserve 6b7568aaea15[585]: File "/usr/local/lib/python3.8/socket.py", line 808, in create_connection
Nov 22 01:38:02 Xserve 6b7568aaea15[585]: raise err
Nov 22 01:38:02 Xserve 6b7568aaea15[585]: File "/usr/local/lib/python3.8/socket.py", line 796, in create_connection
Nov 22 01:38:02 Xserve 6b7568aaea15[585]: sock.connect(sa)
Nov 22 01:38:02 Xserve 6b7568aaea15[585]: TimeoutError: [Errno 110] Operation timed out
Nov 22 01:38:02 Xserve 6b7568aaea15[585]:
Nov 22 01:38:02 Xserve 6b7568aaea15[585]: During handling of the above exception, another exception occurred:
Nov 22 01:38:02 Xserve 6b7568aaea15[585]:
Nov 22 01:38:02 Xserve 6b7568aaea15[585]: Traceback (most recent call last):
Nov 22 01:38:02 Xserve 6b7568aaea15[585]: File "/usr/local/lib/python3.8/site-packages/pyShelly/firmware.py", line 29, in _http_get
Nov 22 01:38:02 Xserve 6b7568aaea15[585]: f = urllib.request.urlopen(url)
Nov 22 01:38:02 Xserve 6b7568aaea15[585]: File "/usr/local/lib/python3.8/urllib/request.py", line 222, in urlopen
Nov 22 01:38:02 Xserve 6b7568aaea15[585]: return opener.open(url, data, timeout)
Nov 22 01:38:02 Xserve 6b7568aaea15[585]: File "/usr/local/lib/python3.8/urllib/request.py", line 525, in open
Nov 22 01:38:02 Xserve 6b7568aaea15[585]: response = self._open(req, data)
Nov 22 01:38:02 Xserve 6b7568aaea15[585]: File "/usr/local/lib/python3.8/urllib/request.py", line 542, in _open
Nov 22 01:38:02 Xserve 6b7568aaea15[585]: result = self._call_chain(self.handle_open, protocol, protocol +
Nov 22 01:38:02 Xserve 6b7568aaea15[585]: File "/usr/local/lib/python3.8/urllib/request.py", line 502, in _call_chain
Nov 22 01:38:02 Xserve 6b7568aaea15[585]: result = func(*args)
Nov 22 01:38:02 Xserve 6b7568aaea15[585]: File "/usr/local/lib/python3.8/urllib/request.py", line 1393, in https_open
Nov 22 01:38:02 Xserve 6b7568aaea15[585]: return self.do_open(http.client.HTTPSConnection, req,
Nov 22 01:38:02 Xserve 6b7568aaea15[585]: File "/usr/local/lib/python3.8/urllib/request.py", line 1353, in do_open
Nov 22 01:38:02 Xserve 6b7568aaea15[585]: raise URLError(err)
Nov 22 01:38:02 Xserve 6b7568aaea15[585]: urllib.error.URLError: <urlopen error [Errno 110] Operation timed out>

**Comment by seb-koch** [comment]
Any updates on this one? Is this a "false positive" or is there an actual issue opening the URL. I got the latest release where the log message issue seems to be fixed but the error persists.

**Comment by hakana** [comment]
It is fixed in the latest beta 0.0.2 Beta 4.
